### ISSUE TITLE
0.24.x breaks sorting for inline tables with overrides for the first key

### ISSUE THREAD

**Issue by vfazio** [pappasam/toml-sort#76]
Our pipelines recently started failing and it looks like its due to this change:

https://github.com/pappasam/toml-sort/commit/9900c719e0e71db8158ed276277367356b79ea07#diff-f893f11eb0bc2f6af62b56b0a01cc76c62e591d06d9979e8212f1cb78d945947R434

https://github.com/pappasam/toml-sort/blame/ba644983fafa863b9fecf450983093efeb93c5de/toml_sort/tomlsort.py#L434

tomlkit's SingleKey's `as_string` seems to _not_ strip the node value, so we end up with stuff like:

```python
        def sort_first(item: TomlSortItem) -> int:
            if item.keys.base in sort_config.first:
                print(item.keys.base)
                print(type(item.keys.base))
                print(item.keys.base.as_string())
                print(len(item.keys.base.as_string()))
                print([ord(z) for z in item.keys.base.as_string()])
                return sort_config.first.index(item.keys.base.as_string())
            return len(sort_config.first)
```

```
(xtf-py3.12) vfazio@vfazio4 ~/development/xtf $ vi /home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py
(xtf-py3.12) vfazio@vfazio4 ~/development/xtf $ pre-commit run -v toml-sort --files pyproject.toml
Sort pyproject.toml......................................................Failed
- hook id: toml-sort
- duration: 0.1s
- exit code: 1

version
<class 'tomlkit.items.SingleKey'>
version
8
[118, 101, 114, 115, 105, 111, 110, 32]
Traceback (most recent call last):
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/bin/toml-sort", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/cli.py", line 405, in cli
    ).sorted()
      ^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 784, in sorted
    sorted_toml = self.toml_doc_sorted(toml_doc)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 770, in toml_doc_sorted
    self.toml_elements_sorted(item, sorted_document),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 583, in toml_elements_sorted
    self.toml_elements_sorted(item, previous_item),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 583, in toml_elements_sorted
    self.toml_elements_sorted(item, previous_item),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 578, in toml_elements_sorted
    for item in self.sorted_children_table(original.keys, original.children):
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 528, in sorted_children_table
    non_tables = self.sort_items(
                 ^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 495, in sort_items
    item.value = self.sort_item(item.keys, item.value)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 419, in sort_item
    return self.sort_inline_table(keys, item, indent_depth=indent_depth)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 460, in sort_inline_table
    tomlsort_items = self.sort_keys(tomlsort_items, sort_config)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 443, in sort_keys
    items = sorted(items, key=sort_first)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vfazio/.cache/pre-commit/repo9__4fg_r/py_env-python3/lib/python3.12/site-packages/toml_sort/tomlsort.py", line 439, in sort_first
    return sort_config.first.index(item.keys.base.as_string())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: 'version ' is not in list
```

This was reproducible via a poetry lock file that pinned a dependency like so:

```
typing-extensions = {version = "4.12.2", python = "<3.12"} # required for TypedDict in python <3.12 per pydantic
```

Seems "fixable" with `as_string.strip()`. I can send a PR if this seems reasonable, but may warrant a test case so it's caught in the future?

@pappasam

**Comment by pappasam** [comment]
Thanks for raising this issue! Can you please send me:

1. Your Python version
2.  your `toml-sort` version?
3. A full toml file that breaks on the latest release, and 
4. The specific cli command that I can run on it to make it break?

I've tried putting the above line in a `pyproject.toml` file, but I haven't been able to reproduce.

Alternatively, if you want to submit a pull request with the fix and a test file, I'm happy to review that as well! Always happy to have a new contributor

**Comment by vfazio** [comment]
Ah yea, key setting was probably missing:

```
(venv) vfazio@vfazio4 /tmp/tmp.ko3XXA2EA6 $ python3 -V
Python 3.12.4

(venv) vfazio@vfazio4 /tmp/tmp.ko3XXA2EA6 $ pip list
Package    Version
---------- -------
pip        22.0.2
setuptools 59.6.0
toml-sort  0.24.1
tomlkit    0.13.2


(venv) vfazio@vfazio4 /tmp/tmp.ko3XXA2EA6 $ cat pyproject.toml

[tool.poetry.dependencies]
typing-extensions = {version = "4.12.2", python = "<3.12"} # required for TypedDict in python <3.12 per pydantic

[tool.tomlsort]
all = true
ignore_case = true
in_place = true
trailing_comma_inline_array = true
overrides."tool.poetry.dependencies".first = ["python"]
overrides."tool.poetry.dependencies.typing-extensions".first = ["version"]

(venv) vfazio@vfazio4 /tmp/tmp.ko3XXA2EA6 $ toml-sort pyproject.toml
Traceback (most recent call last):
  File "/tmp/tmp.ko3XXA2EA6/venv/bin/toml-sort", line 8, in <module>
    sys.exit(cli())
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/cli.py", line 405, in cli
    ).sorted()
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 779, in sorted
    sorted_toml = self.toml_doc_sorted(toml_doc)
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 765, in toml_doc_sorted
    self.toml_elements_sorted(item, sorted_document),
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 578, in toml_elements_sorted
    self.toml_elements_sorted(item, previous_item),
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 578, in toml_elements_sorted
    self.toml_elements_sorted(item, previous_item),
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 573, in toml_elements_sorted
    for item in self.sorted_children_table(original.keys, original.children):
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 523, in sorted_children_table
    non_tables = self.sort_items(
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 490, in sort_items
    item.value = self.sort_item(item.keys, item.value)
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 419, in sort_item
    return self.sort_inline_table(keys, item, indent_depth=indent_depth)
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 455, in sort_inline_table
    tomlsort_items = self.sort_keys(tomlsort_items, sort_config)
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 438, in sort_keys
    items = sorted(items, key=sort_first)
  File "/tmp/tmp.ko3XXA2EA6/venv/lib/python3.10/site-packages/toml_sort/tomlsort.py", line 434, in sort_first
    return sort_config.first.index(item.keys.base.as_string())
ValueError: 'version ' is not in list

```

Dropping that override for typing-extensions seems to also work around it. I'm pretty sure we added that for a reason and i think it was so the version key was the first key in the poetry dependency list for easy diff review.

**Comment by vfazio** [comment]
i'll try to submit a PR once i get back into the office. We're not currently in a broken state as i've pinned toml-sort to 0.23.1

**Comment by vfazio** [comment]
looks like there's already a test for overrides and "first", so i wonder if there's some subtlety in toml-kit about json ... The JSON examples in the tests do not have space separators between the KvP.

So, interestingly, the following works:

```

[tool.poetry.dependencies]
typing-extensions = {version="4.12.2", python = "<3.12"} # required for TypedDict in python <3.12 per pydantic

```
However, a difference is detected because the KvP is readded with spaces

```
(venv) vfazio@vfazio4 /tmp/tmp.ko3XXA2EA6 $ toml-sort -o test pyproject.toml

(venv) vfazio@vfazio4 /tmp/tmp.ko3XXA2EA6 $ diff test pyproject.toml
6,7c6,7
< # typing-extensions = {version="4.12.2", python = "<3.12"} # required for TypedDict in python <3.12 per pydantic
< typing-extensions = {version = "<3.12", python = "4.12.2"} # required for TypedDict in python <3.12 per pydantic
---
> #typing-extensions = {version="4.12.2", python = "<3.12"} # required for TypedDict in python <3.12 per pydantic
> typing-extensions = {python="4.12.2", version="<3.12"} # required for TypedDict in python <3.12 per pydantic
```

Rerunning toml-sort on the result breaks because of those spaces... so i'm inclined to think they do indeed need to be stripped or toml-kit's original value should be the stripped version. I'll still putz with it when i have a real setup to debug further

**Comment by vfazio** [comment]
Just as a note:

changing the line back to `return sort_config.first.index(item.keys.base)` seems to work in my specific case. My guess is that when comparing values via `index`, it's using `__eq__` or `__hash__` which uses the `SingleKey.key` value, where as `as_string` returns the unstripped `SingleKey._original` value.

which is "correct", i don't know yet. I'm not sure why this line was changed to begin with as the commit log isn't very clear, but maybe it was due to existing tests failing?

**Comment by vfazio** [comment]
@pappasam i made an attempt to fix this in #77, lmk your thoughts
